bin/lib/homepage.js

```diff
@@ -0,0 +1,126 @@
+require('dotenv').config();
+const fetch = require('node-fetch');
+const { sections } = require('./page-structure');
+const structure = require('./page-structure');
+const Utils = require('./utils');
+
+
+async function getAllImages(edition = 'uk') {
+	let allSectionImages = [];
+	
+	for(let i = 0; i < sections.length; ++i) {
+		const sectionData = await getList(sections[i][edition], sections[i].isConcept);
+		let layout = sectionData.hasOwnProperty('layoutHint')?sectionData.layoutHint:sections[i].layout;
+		
+		const sectionImages = await getImagesFor(sectionData.items, structure.getPositions(layout));
+		allSectionImages = allSectionImages.concat(sectionImages);
+
+		if(i === 0 && layout === 'landscape') {
+			/*Special accomodation for when landscape piece is opinion*/
+			const sectionHeadshots = await getHeadshotsFor(sectionData.items, 2);
+			allSectionImages = allSectionImages.concat(sectionHeadshots);	
+		} else if(sections[i].checkHeadshots !== null) {
+			const sectionHeadshots = await getHeadshotsFor(sectionData.items, sections[i].checkHeadshots);
+			allSectionImages = allSectionImages.concat(sectionHeadshots);
+		}
+	}
+
+	console.log(allSectionImages.length);
+	return allSectionImages;
+}
+
+async function getImagesFor(list, indices) {
+	const links = [];
+	for(let i = 0; i < indices.length; ++i) {
+		const imageData = await getTeaser(Utils.extractUUID(list[indices[i]]));
+
+		if(imageData.length) {
+			links.push(imageData[0].binaryUrl.replace(process.env.API_IMG_URL, process.env.REPLACE_IMG_URL).concat('?source=janetbot'));	
+		}
+	}
+
+	return links;
+}
+
+async function getTeaser(uuid) {
+	console.log('getTeaser::', uuid);
+	return fetch(`http://api.ft.com/enrichedcontent/${uuid}?apiKey=${process.env.FT_API_KEY}`)
+			.then(res => res.json())
+			.then(data => {
+				if(data.alternativeImages && data.alternativeImages.promotionalImage) {
+
+					return new Array(data.alternativeImages.promotionalImage);
+				}
+
+				if(data.mainImage) {
+					return data.mainImage.members;	
+				}
+
+				return [];
+				
+			})
+			.catch(err => { throw err });
+}
+
+async function getAuthor(uuid) {
+	console.log('getAuthor::', uuid);
+	return fetch(`http://api.ft.com/enrichedcontent/${uuid}?apiKey=${process.env.FT_API_KEY}`)
+			.then(res => res.json())
+			.then(data => { return data.annotations })
```

Comment:
minor point: different syntax doing same thing. Is it better to use the abbreviated form, res => res.json(), or the more explicit form, res => { return res.json() }
?

Reply:
I normally use `return` only on the last statement in the promise chain (or when there are computations to be made.